src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1191,13 +1191,158 @@ class C
+            //         var attr = new System.Runtime.CompilerServices.RequiresLocationAttribute();
+            Diagnostic(ErrorCode.ERR_BadAccess, "RequiresLocationAttribute").WithArguments("System.Runtime.CompilerServices.RequiresLocationAttribute").WithLocation(5, 56));
+    }
+
+    [Fact]
+    public void FunctionPointer_DefineManually_LaterDefinedInRuntime()
+    {
+        // Library defines an API with function pointer, manually declaring the attribute.
+        var source1 = """
+            public class C
+            {
+                public unsafe void M(delegate*<ref readonly int, void> f)
+                {
+                    int x = 123;
+                    f(in x);
+                }
+            }
+            """;
+        var comp1v1 = CreateCompilation(new[] { source1, RequiresLocationAttributeDefinition }, assemblyName: "Assembly1", options: TestOptions.UnsafeReleaseDll);
+        comp1v1.VerifyDiagnostics();
+        var comp1v1Ref = comp1v1.EmitToImageReference();
+
+        // Consumer can use the API.
+        var source2 = """
+            public class D
+            {
+                public unsafe void M()
+                {
+                    new C().M(&F);
+                }
+                static void F(ref readonly int x) => System.Console.Write("F" + x);
+            }
+            """;
+        var comp2 = CreateCompilation(source2, new[] { comp1v1Ref }, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics();
+        var comp2Ref = comp2.EmitToImageReference();
+
+        var source3 = """
+            try
+            {
+                new D().M();
+            }
+            catch (System.MissingMethodException e)
+            {
+                System.Console.Write(e.Message);
+            }
+            """;
+        CompileAndVerify(source3, new[] { comp1v1Ref, comp2Ref }, expectedOutput: "F123").VerifyDiagnostics();
+
+        // .NET runtime declares the attribute.
+        var source4 = """
+            namespace System.Runtime.CompilerServices
+            {
+                [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+                public sealed class RequiresLocationAttribute : Attribute
+                {
+                }
+            }
+            """;
+        var comp4 = CreateCompilation(source4).VerifyDiagnostics();
+        var comp4Ref = comp4.EmitToImageReference();
+
+        // Library is recompiled against the newest runtime.
+        var comp1v2 = CreateCompilation(source1, new[] { comp4Ref }, assemblyName: "Assembly1", options: TestOptions.UnsafeReleaseDll);
+        comp1v2.VerifyDiagnostics();
+        var comp1v2Ref = comp1v2.EmitToImageReference();
+
+        // That breaks the consumer.
+        CompileAndVerify(source3, new[] { comp1v2Ref, comp2Ref, comp4Ref },
+            expectedOutput: "Method not found: 'Void C.M(Void (Int32 ByRef))'.").VerifyDiagnostics();
+    }
+
```

Comment:
Would it be good if we had 1 more test, which verifies that if you declare it locally, but the runtime also provide one, then it uses your local one?